### ISSUE TITLE
DevDocs tells me "AppCache isn't available" on Firefox 35 / Mac OS X 10.9.5

### ISSUE THREAD

**Issue by nhoizey** [Thibaut/devdocs#155]
![2015-01-21 at 16 12](https://cloud.githubusercontent.com/assets/78213/5838636/1c50b7c0-a189-11e4-9d6b-ddfdc01b3ac4.png)

**Comment by Thibaut** [comment]
Are you running DevDocs in a private window?
Did you allow DevDocs to store data offline? (Firefox should have prompted you when you first visited the site)

**Comment by nhoizey** [comment]
I'm not in a private window and Firefox didn't prompt me.

**Comment by bassu** [comment]
@nhoizey Use Firefox Developer Edition. I have it installed and am currently on Aurora channel (37.0a2) and it works quite well (though I'm using OSX Yosemite).

**Comment by nhoizey** [comment]
@bassu stable Firefox supports app cache, I don't want to change my browsing habits just because one service doesn't support it… ;-)

**Comment by nhoizey** [comment]
(sorry, closed it by error)

@bassu stable Firefox supports app cache, I don't want to change my browsing habits just because one service doesn't support it… ;-)

**Comment by bassu** [comment]
@nhoizey I don't think stable Firefox is good enough for developers at this time. There have been several bug reports with DOM and HTML5 specifically in Firefox while it tries to keep up with Chrome. I am sure there's some bug somewhere which is rendering appcache useless in stable Firefox.

Browser is a fast moving target and Firefox has some history of coming late to party (although they are trying to change it with Dev Edition).

**Comment by nhoizey** [comment]
@bassu OMG, do you really mean all this? This is clear FUD.

**Comment by julienw** [comment]
Firefox supports app cache properly for ages. Would be nice if you tell exactly what's missing for you.

**Comment by julienw** [comment]
That said, I'm running stable iceweasel (which is Firefox 31) and it seems to work for me?

**Comment by kud** [comment]
Because it's a dev doc, it should be used with a "dev" browser? Come on, be smarter.

**Comment by Thibaut** [comment]
@nhoizey sorry I don't know what's going on here. I use Firefox and AppCache works fine.

Can you check that `browser.cache.offline.enable` is set to `true` (double-click to toggle value) in `about:config`?

**Comment by nhoizey** [comment]
@Thibaut it is set to ```true```

**Comment by BenoitZugmeyer** [comment]
[This](https://github.com/Thibaut/devdocs/blob/master/assets/javascripts/app/appcache.coffee#L4) is the test to check if appcache is available on your browser. Could you check the result of `applicationCache.status !== applicationCache.UNCACHED` in your console (while browsing devdocs.io) ?

**Comment by Thibaut** [comment]
@nhoizey try:

- Preferences -> Advanced -> Network -> Offline Web Content and User Data
  - Click "Clear Now"
  - Remove "devdocs.io" from the list if you see it
  - Make sure DevDocs isn't in the list of "Exceptions…"
- Clear Recent History -> Everything -> Only check "Site Preferences"

and then restart Firefox, go to devdocs.io, wait a few seconds, reload, and check the offline page.

**Comment by nhoizey** [comment]
@BenoitZugmeyer it says ```true```

**Comment by nhoizey** [comment]
@Thibaut it worked, thanks!

**Comment by Thibaut** [comment]
@nhoizey Great!

Closing… unfortunately there's nothing I can do on DevDocs's side to prevent Firefox from getting into this state.

**Comment by julienw** [comment]
@nhoizey any extension that could change this? Can you try with a clean profile?

**Comment by julienw** [comment]
mm sorry I missed the last messages :p

**Comment by nhoizey** [comment]
@julienw :-D

**Comment by julienw** [comment]
That said, I have the same issue on Firefox OS. Yet it works fine when I enable "flight mode" to trigger the offline mode.

Can you explain what is this "state" you're talking about in https://github.com/Thibaut/devdocs/issues/155#issuecomment-72212707 ?

(note: I'm a huge fan of devdocs.io)

**Comment by Thibaut** [comment]
@julienw I have no idea what's causing this, sorry. On desktop I managed to lock AppCache by playing with the settings and clearing the cache, but I can't replicate that now…

**Comment by julienw** [comment]
OK ... Anyway this works fine, only the FAQ displays bad information.

**Comment by myfonj** [comment]
Ran into this situation as well with Nightly. After clearing caches and still stuck with `…isn't available (or disabled)…` I realized I was at `http://` all the time and that `applicationCache` is restricted for `https://` only (nowadays). After switching to https I got few "application failed to load" screens but after few force reloads it catched and now seems to work properly, hooray.

Anyways, seems it might be worth mentioning info from this issue it in the message. And perhaps force HTTPS (?).
